app/dv/src/main/java/io/syndesis/dv/server/endpoint/DataVirtualizationService.java

```diff
@@ -1065,38 +1100,103 @@ public PodMetrics getPublishedVirtualizationMetrics(
 
         try (SyndesisHttpClient client = new SyndesisHttpClient()){
-        try (InputStream response = client.executeGET(baseUrl + "org.teiid:type=Runtime/Sessions", authHeader);) {
-            ObjectMapper mapper = new ObjectMapper();
-            JsonNode root = mapper.readTree(response);
-            JsonNode value = root.withArray("value");
-            if (value != null) {
-                int sessionCount = value.size();
-                metrics.setSessions(sessionCount);
+            try (InputStream response = client.executeGET(baseUrl + "org.teiid:type=Runtime/Sessions", authHeader);) {
+                ObjectMapper mapper = new ObjectMapper();
+                JsonNode root = mapper.readTree(response);
+                JsonNode value = root.withArray("value");
+                if (value != null) {
+                    int sessionCount = value.size();
+                    metrics.setSessions(sessionCount);
+                }
             }
-        }
 
-        try (InputStream response = client.executeGET(baseUrl + "org.teiid:type=Runtime/TotalRequestsProcessed", authHeader);) {
-            ObjectMapper mapper = new ObjectMapper();
-            JsonNode root = mapper.readTree(response);
-            JsonNode value = root.get("value");
-            if (value != null) {
-                long requestCount = value.asLong();
-                metrics.setRequestCount(requestCount);
+            try (InputStream response = client.executeGET(baseUrl + "org.teiid:type=Runtime/TotalRequestsProcessed", authHeader);) {
+                ObjectMapper mapper = new ObjectMapper();
+                JsonNode root = mapper.readTree(response);
+                JsonNode value = root.get("value");
+                if (value != null) {
+                    long requestCount = value.asLong();
+                    metrics.setRequestCount(requestCount);
+                }
             }
-        }
 
-        try (InputStream response = client.executeGET(baseUrl + "org.teiid:type=Cache,name=ResultSet/HitRatio", authHeader);) {
-            ObjectMapper mapper = new ObjectMapper();
-            JsonNode root = mapper.readTree(response);
-            JsonNode value = root.get("value");
-            if (value != null) {
-                double hitRatio = value.asDouble();
-                metrics.setResultSetCacheHitRatio(hitRatio);
+            try (InputStream response = client.executeGET(baseUrl + "org.teiid:type=Cache,name=ResultSet/HitRatio", authHeader);) {
+                ObjectMapper mapper = new ObjectMapper();
+                JsonNode root = mapper.readTree(response);
+                JsonNode value = root.get("value");
+                if (value != null) {
+                    double hitRatio = value.asDouble();
+                    metrics.setResultSetCacheHitRatio(hitRatio);
+                }
             }
         }
-        }
 
         return metrics;
     }
 
+    @GetMapping(value = {
```

Comment:
So there is no API for to get View level for getting a role? Maybe it is already on some getView?

Reply:
No, based upon all of our conversations that was not needed yet.